### ISSUE TITLE
binding RAPI to a specific IP makes the cron watcher unhappy

### ISSUE THREAD

**Issue by Ganeti-Issues-Migrator** [ganeti/ganeti#1170]
Originally reported of Google Code with ID 1114.

```
> What software version are you running?

Ganeti from the PPA "pkg-ganeti-devel".

> Please provide the output of "gnt-
cluster --version", "gnt-cluster version", and "hspace --version".

# gnt-cluster --version
gnt-cluster (ganeti v2.11.6) 2.11.6
# gnt-cluster version
Software version: 2.11.6
Internode protocol: 2110000
Configuration format: 2110000
OS api version: 20
Export interface: 0
VCS version: (ganeti) version v2.11.6


> What distribution are you using?

Ubuntu 14.04 64 bit with KVM and DRBD

> What steps will reproduce the problem?

1. Set RAPI_ARGS="-b $CLUSTER_IP" in /etc/default/ganeti
2. Restart the Ganeti services
3. Call the ganeti-watcher cron
4. Check the watcher's logs (/var/log/ganeti/watcher.log)

> What is the expected output?

I would expect no error/warning.

> What do you see instead?

2015-07-01 12:15:01,282: ganeti-watcher pid=4091 INFO RunCmd /usr/lib/ganeti/daemon-util check-and-start ganeti-confd
2015-07-01 12:15:01,288: ganeti-watcher pid=4091 INFO RunCmd /usr/lib/ganeti/daemon-util check-and-start ganeti-mond
2015-07-01 12:15:01,294: ganeti-watcher pid=4091 INFO RunCmd /usr/lib/ganeti/daemon-util check-and-start ganeti-kvmd
2015-07-01 12:15:01,411: ganeti-watcher pid=4091 INFO RunCmd /usr/lib/ganeti/daemon-util check-and-start ganeti-rapi
2015-07-01 12:15:01,417: ganeti-watcher pid=4091 WARNING RAPI error: (7, 'Failed to connect to 127.0.0.1 port 5080: Connection refused')
2015-07-01 12:15:01,417: ganeti-watcher pid=4091 WARNING Couldn't get answer from remote API, restaring daemon
2015-07-01 12:15:01,417: ganeti-watcher pid=4091 INFO RunCmd /usr/lib/ganeti/daemon-util stop ganeti-rapi
2015-07-01 12:15:01,443: ganeti-watcher pid=4091 INFO RunCmd /usr/lib/ganeti/daemon-util check-and-start ganeti-rapi
2015-07-01 12:15:01,663: ganeti-watcher pid=4091 WARNING RAPI error: (7, 'Failed to connect to 127.0.0.1 port 5080: Connection refused')
2015-07-01 12:15:01,663: ganeti-watcher pid=4091 CRITICAL RAPI is not responding
2015-07-01 12:15:01,771: ganeti-watcher pid=4153 INFO Watcher for node group 'd031ad03-3232-44fe-b58d-f0985ba0a2b1'
2015-07-01 12:15:01,771: ganeti-watcher pid=4153 ERROR Can't acquire lock on state file /var/lib/ganeti/watcher.d031ad03-3232-44fe-b58d-f0985ba0a2b1.data: File already locked


> Please provide any additional information below.

Telling RAPI to use a specific interface through "-i" doesn't help. Only binding to 0.0.0.0 works around the issue.
```
Originally added on 2015-07-01 16:18:56 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
I could reproduce this and will have a look for a fix.

I'm curious though, what was your motivation to use this option?
```
Originally added on 2015-07-02 08:47:55 +0000 UTC.

Added Labels: Type-Defect Priority-Medium
Changed State: Accepted
Added to Milestone: Release2.12

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2015-07-02 14:11:57 +0000 UTC.

Changed State: Started

**Comment by Ganeti-Issues-Migrator** [comment]
```
Great, thanks.

To not expose RAPI to the secondary network (for DRBD replication). For
now we use IPtables.
```
Originally added on 2015-07-02 14:23:14 +0000 UTC.

**Comment by Ganeti-Issues-Migrator** [comment]
```
Fixed in 2.12 with:

commit fbf8089c7f19d6c4d9844d00d2f25a71f644a5b0
Author: Helga Velroyen <helgav@google.com>
Date:   Thu Jul 2 13:38:17 2015 +0200

    Watcher: add option for setting RAPI IP
    
    Per default, the RAPI daemon binds to 0.0.0.0 when being
    started. This means it serves from any IP the machine is
    configured for. This works well together with the watcher
    which always polls the RAPI daemons on 127.0.0.1 and
    restarts it when it is not reachable.
    
    If a user decides to start the RAPI daemon with a particular
    IP other than 127.0.0.1 (using the option -b, for example
    set in /etc/default/ganeti), RAPI will only serve from that
    IP and thus it will not be reachable from 127.0.0.1. Since
    the watcher only polls on this IP, it will inevitably fail
    to connect to the RAPI daemon and thus restart it every five
    minutes.
    
    To solve this, this patch adds an option --rapi-ip to the
    watcher. Whenever -b of the RAPI daemon is set, the watcher
    needs to be fed the same IP with --rapi-ip (which means
    editing /etc/cron.d/ganeti for example). This is not optimal
    regarding user experience (as it is easy to forget one of
    the two places), but the alternative would be to make this
    a ganeti configuration parameter which is fed to both, RAPI
    daemon and watcher, but this would be significantly more
    effort for this relatively rarely used feature.
    
    Signed-off-by: Helga Velroyen <helgav@google.com>
    Reviewed-by: Petr Pudlak <pudlak@google.com>

```
Originally added on 2015-07-03 10:07:07 +0000 UTC.

Changed State: Fixed

**Comment by Ganeti-Issues-Migrator** [comment]
```
-- Empty comment --
```
Originally added on 2015-07-13 17:13:17 +0000 UTC.

Changed State: Released
